map_with_sidebar.css

```diff
@@ -1,22 +1,77 @@
-/* This is for a page with a fullscreen map and a collapsible left sidebar */
+/* This is for a page with a map, collapsable sidebar, and collapsable bottom form area. */
 
-/* The map */
+/* Defaults */
+* {
+  box-sizing: border-box;
+}
 body {
-  margin: 0;
-  padding: 0;
+  font: 16px Helvetica Nueu,Arial,Helvetica,sans-serif;
 }
 
-#map {
+/* The application layout */
+body {
+  margin: 0;
+  padding: 0;
   position: absolute;
   top: 0;
+  left: 0;
+  right: 0;
   bottom: 0;
-  width: 100%;
+
+  overflow: hidden;
+
+  /* Two columns, the sidebar on the left, and the main content on the right. */
+  display: flex;
+  flex-flow: row nowrap;
+}
+
+body > aside {
+  position: relative; /* for children who want to position:absolute */
+
+  flex: 0;
+  min-width: 25rem; /* use min/max-width because we're in a flex containter. */
+  max-width: 25rem;
+}
+.sidebar-content {
+  width: 25em; /* Keep the size constant, so the map overlaps it when it collapses. */
+
+  max-height: 100%;
+  overflow-y: auto;
+}
+body > aside.collapsed {
+  min-width: 1.5rem;
+  max-width: 1.5rem;
+}
+
+body > main {
+  flex-grow: 1;
+
+  /* Potentially split in two: map on the top, form on the bottom. */
+  display: flex;
+  flex-flow: column nowrap;
+}
+
+#map {
+  position: relative; /* for children who want to position:absolute */
+
+  flex-grow: 1;
+}
+
+#panel {
+  flex-grow: 0;
+  max-height: 50vh;
+  overflow-y: auto;
+}
+#panel:empty {
+  display: none;
 }
 
+/* The map */
 #basemaps-wrapper {
+  z-index: 1;
   position: absolute;
-  bottom: 100px;
-  right: 20px;
+  bottom: 80px;
```

Comment:
In A/B Street-land, we've been careful about things based on pixels, because they scale with a monitor's DPI (Mac Retina displays have been a source of endless bugs). Playing around with my OS's display scale factor, things look reasonable, so it's cool to see that browsers handle this pretty well

Reply:
"CSS pixels" are a whole thing. They started out defined as representing device pixels, but the definition changed to ... (looking up definition)... 1/96th of 1 inch, sure.

But the long and short of it is, they are just another unit of "absolute length" in CSS land, that will be scaled to look similar in different environments, so you can use them naively.